File: Diego Rafael Ramos Llanos/java/leccion7/leccion7/src/leccion7/ejercicio2.java
/*Ejercicio 12: Pedir un número y calcular su factorial
Hacerlo con las dos clases, Scanner y JOptionPane*/
package leccion7;

//import java.util.Scanner;
import javax.swing.JOptionPane;

public class ejercicio2 {
        public static void main(String[] args) {
        //Scanner scanner = new Scanner(System.in);
        int numero=Integer.parseInt(JOptionPane.showInputDialog("Ingresa un número para calcular su factorial: "));
       // System.out.print("Ingresa un número para calcular su factorial: ");
        //int numero = scanner.nextInt();
       // scanner.close();//cerrar scaner es una buena practica

        long factorial = 1;

        // Calcula el factorial utilizando un bucle for
        for (int i = 1; i <= numero; i++) {
            factorial *= i;
        }
        JOptionPane.showMessageDialog(null,"El factorial de " + numero + " es " + factorial);
       // System.out.println("El factorial de " + numero + " es " + factorial);
    }
}

File: Diego Rafael Ramos Llanos/python/leccion 5/ejercicio2.py
# ejercicio 5: Factorial de un número positivo
# Hacer un programa para calcular el factorial de un número positivo
numero = int(input("Digite un numero positivo: "))
factorial=1
while numero < 0:
    print("El número debe ser positivo.")
    numero = int(input("Digite un numero positivo: "))
if numero>0:
    for i in range (1,numero+1):
        factorial*=i

print(f"El factorial de {numero} es {factorial}")